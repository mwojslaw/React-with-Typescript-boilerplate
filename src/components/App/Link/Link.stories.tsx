import * as React from 'react'
import { storiesOf } from '@storybook/react'
import Link from './';

storiesOf('Link', module)
  .add('Link', () => <Link href={''} text={'Sample text'} />)