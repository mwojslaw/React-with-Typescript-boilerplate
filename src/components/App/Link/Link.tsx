import * as React from 'react';

const Link = ({href, text}: {href: string, text: string}) => (
    <a href={href}>{text}</a>
);

export default Link;