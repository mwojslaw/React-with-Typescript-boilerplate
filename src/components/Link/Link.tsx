import * as React from 'react';

const styles = require('./Link.css');

const Link = ({href, children}: {href: string, children: string}) => (
    <a className={styles.Link} href={href}>{children}</a>
);

export default Link;