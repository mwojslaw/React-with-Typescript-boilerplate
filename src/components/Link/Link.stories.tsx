import * as React from 'react'
import { storiesOf } from '@storybook/react'
import Link from './';

storiesOf('Link', module)
  .add('Link', () => <Link href={''}>Sample text</Link>)