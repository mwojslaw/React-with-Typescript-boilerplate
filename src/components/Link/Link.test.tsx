import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Link from "./";

test("Link should render", () => {
    const component = renderer.create(
        <Link href={''}>React</Link>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});