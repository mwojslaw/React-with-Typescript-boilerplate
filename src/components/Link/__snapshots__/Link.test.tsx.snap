// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render 1`] = `
<a
  className="Link"
  href=""
>
  React
</a>
`;
